go-controller/pkg/retry/obj_retry.go

```diff
@@ -375,7 +379,7 @@ func (r *RetryFramework) periodicallyRetryResources() {
 			r.iterateRetryResources()
 			timer.Reset(RetryObjInterval)
 
-		case <-r.StopChan:
+		case <-waitCh:
```

Comment:
Thanks Ben for fixing this! I have one concern, though.
So the missing piece was that we were not waiting on the stop channel in the watchFactory. However, now we're not waiting on the stop channel from ovnk master (and soon, ovnk node!) that we pass as input when creating a new retry framework: https://github.com/ovn-org/ovn-kubernetes/blob/e3a780672e8dae91363d277c324db760c3d36f4d/go-controller/pkg/ovn/obj_retry_master.go#L61-L65

 Should we have a `case` statement for both channels now?

Reply:
On a second thought,  when we end ovnkube master or node processes, we first close their respective channel and then proceed to close the `watchFactory` channel. 
https://github.com/ovn-org/ovn-kubernetes/blob/e3a780672e8dae91363d277c324db760c3d36f4d/go-controller/cmd/ovnkube/ovnkube.go#L306-L311
So, unless I'm missing some other cases, it looks OK to only wait on the `watchfactory`channel. In this case, we should get rid of the stop channel in the retry framework, which we wouldn't use any longer. What do you think, @trozet?

Reply:
That was indeed out thinking, @ricky-rav . Waiting on the stopChannel of the watchFactory should be enough (and better), since ovnkube would be the ones that shutdown the watchFactory. @trozet 's credit in teaching us that. ☺️